### PR TITLE
fix(sidebar): remove deleted workspace from sidebar and handle re-delete gracefully

### LINKED ISSUE #374: bug(sidebar): deleted workspace persists in sidebar and shows 'Workspace not found' on re-delete
## Summary

After deleting a workspace, it still appears in the sidebar. Attempting to delete it again shows a "Workspace not found" error in the delete confirmation modal, since the record has already been removed from the database but not from the frontend state.

## Steps to Reproduce

1. Delete a workspace (click trash icon → confirm in modal)
2. The backend deletes the workspace from the database successfully
3. The workspace remains visible in the sidebar
4. Click the trash icon on the same workspace again
5. The "Delete workspace" modal opens, but clicking Delete shows the error "Workspace not found" (see screenshot)

### 1. Sidebar not removing the workspace after deletion

When `deleteWorkspace` succeeds in `DeleteWorkspaceModal.tsx`, it calls `removeWorkspace(wsId)` on the Zustand store. Either:
- `removeWorkspace` is not properly filtering the workspace out of the `workspaces` array in the store
- The store update isn't triggering a re-render of the sidebar
- There's a race condition where the workspace list gets refreshed/overwritten after the removal (e.g., a periodic sync or event handler re-adding stale data)

### 2. "Workspace not found" error shown to user

When a workspace has already been deleted from the DB, the backend's `delete_workspace` command returns an error string "Workspace not found". The modal displays this raw error to the user. Instead, this case should be handled gracefully — if the workspace is already gone, treat it as a success: remove it from the store and close the modal.

## Expected Behavior

1. After a successful delete, the workspace should immediately disappear from the sidebar
2. If a delete is attempted on an already-deleted workspace (edge case), the frontend should silently clean it up from the store rather than showing an error

## Key Files

- `src/ui/src/components/modals/DeleteWorkspaceModal.tsx` — calls `deleteWorkspace` then `removeWorkspace`
- `src/ui/src/components/sidebar/Sidebar.tsx` — renders workspace list from store
- `src-tauri/src/commands/workspace.rs` → `delete_workspace` — returns "Workspace not found" error
- Zustand store `removeWorkspace` action — should filter workspace from state

### PR DESCRIPTION
**Pull request by doomspork**
## Summary

Fixes a bug where deleted workspaces persist in the sidebar and show a "Workspace not found" error when re-deleted.

- **Backend**: Make `delete_workspace` idempotent — return `Ok(())` if the workspace is already gone instead of erroring. Also handle missing repository gracefully by skipping git cleanup while still performing DB, agent, and tray cleanup.
- **Frontend**: Always remove the workspace from the Zustand store after the user confirms deletion, regardless of backend errors. The user's intent is unambiguous and the store is ephemeral (rebuilt from DB on launch).

Closes #374

## Complexity Notes

The backend change makes the repository lookup non-fatal (`find` returns `Option` instead of `ok_or`). Git operations (worktree removal, branch deletion) are conditionally skipped when the repo is missing, but DB cascade, agent cleanup, MCP supervisor cleanup, and tray rebuild still run. This handles the edge case where a workspace exists but its parent repository was already deleted.

## Test Steps

1. Create a workspace, then delete it → it should immediately disappear from the sidebar
2. Verify no "Workspace not found" errors appear in any scenario
3. Run `cargo test -p claudette -- test_delete_workspace_idempotent` → passes (DB-level idempotency)
4. Run `cd src/ui && bun run test` → all tests pass including new `removeWorkspace` store tests

## Checklist

- [x] Tests added/updated
- [x] Documentation updated (if applicable)

### CODE REVIEW

File: src/ui/src/stores/useAppStore.test.ts [hunk 1]
```diff
@@ -1326,3 +1327,63 @@ describe("rollbackConversation updates latestTurnUsage", () => {
...
+
+describe("removeWorkspace", () => {
+  beforeEach(() => {
+    useAppStore.setState({
+      workspaces: [makeWorkspace("ws-a"), makeWorkspace("ws-b")],
+      selectedWorkspaceId: "ws-a",
+      unreadCompletions: new Set(["ws-a", "ws-b"]),
+      terminalTabs: { "ws-a": [{ id: "t1", label: "shell" }], "ws-b": [{ id: "t2", label: "shell" }] },
+      activeTerminalTabId: { "ws-a": "t1", "ws-b": "t2" },
+      workspaceTerminalCommands: { "ws-a": ["ls"], "ws-b": ["pwd"] },
```
[the remark this answers is not in the document]
Comment (doomspork) [reply src/ui/src/stores/useAppStore.test.ts]: Done — updated fixtures to use the complete `TerminalTab` shape (`id: number`, `workspace_id`, `title`, `is_script_output`, `sort_order`, `created_at`) and proper `WorkspaceCommandState` objects. See commits c7cf31b and e76083c.

File: src/ui/src/components/modals/DeleteWorkspaceModal.tsx [hunk 2]
```diff
@@ -18,12 +17,13 @@ export function DeleteWorkspaceModal() {
...
-      removeWorkspace(wsId);
-      closeModal();
-    } catch (e) {
-      setError(String(e));
-      setLoading(false);
+    } catch {
+      // Backend may fail (already deleted, DB error, etc.). Proceed with
+      // local removal — the user confirmed deletion and the workspace will
+      // reappear on next launch if the DB row actually survived.
     }
```
[the remark this answers is not in the document]
Comment (doomspork) [reply src/ui/src/components/modals/DeleteWorkspaceModal.tsx:22]: Done — the catch block now captures the error and logs it via `console.error`. See commit 71aabfc.
